### ISSUE TITLE
Timer should have fired at ... which is ~3 hours ago

### ISSUE THREAD

**Issue by oleggolovkovuss** [dotnet/orleans#6410]
Hello, my silos seem to be doing fine however I keep getting this warning in logs every ~5 minutes along with the other message saying 

> Watchdog had 1 Health Check Failure(s) out of 5 Health Check Participants.

Should I do something about it? Should it be self-healing? Reboot helps for a couple of days

My setup:
Orleans 3.0
ASP.NET Core 3.0 in Docker on Linux
ADO.NET clustering (MySQL)
3 Silos total, 1 week uptime for each

**Comment by veikkoeeva** [comment]
@oleggolovkovuss To check, is your system, and the siloes, still functional when you reboot? Does this issues reoccur every couple of days and then continue for every five minutes once it has started?

I include here for the benefit of others that the error messages originates from https://github.com/dotnet/orleans/blob/master/src/Orleans.Runtime/Silo/Watchdog.cs.

**Comment by oleggolovkovuss** [comment]
> is your system, and the siloes, still functional when you reboot?
It is functional before, during (I use rolling update with 15seconds delay between each silo deploy) and after the reboot.

> Does this issue re-occur every couple of days and then continue for every five minutes once it has started?

Yes, for about 3 or 4 days it's fine and then it starts appearing every 5 minutes until the reboot/redeploy

**Comment by DiAifU** [comment]
Hello,

I'm getting the same warning, but only since Orleans 3.1.x. I didn't have it on silos running 3.0.2. 
Which version are you using ?

**Comment by oleggolovkovuss** [comment]
Hello, I'm using 3.0.0

**Comment by DiAifU** [comment]
Hmm, so probably not the same problem as mine.
Just so you know, if you're using Localhost clustering, then it seems to be normal (#6218)

**Comment by ReubenBond** [comment]
I think I see the issue, and I'll open a PR to fix it soon - apologies for the hassle. It is almost certainly benign, but to be sure: do any of you see instances of these messages in your logs?:

* `Exception while monitoring cluster members`
* `Failed to update table entry for this silo`
* `Failed to clean up defunct membership table entries`
* `CleanupDefunctSiloEntries operation is not supported by the current implementation of` (I believe this is the issue - this timer needs to be disposed after this message is logged)
* `Error refreshing membership table`

**Comment by DiAifU** [comment]
In my case I have this message at the same time : 
`Watchdog had 1 Health Check Failure(s) out of 6 Health Check Participants.`

**Comment by ReubenBond** [comment]
@DiAifU do you see any of those other messages that I listed?

**Comment by DiAifU** [comment]
> 
> 
> @DiAifU do you see any of those other messages that I listed?

No, only these two :
`Timer should have fired at 03/20/2020 14:44:27, which is 01:11:29.8424276 ago` and
`Watchdog had 1 Health Check Failure(s) out of 6 Health Check Participants.`

**Comment by oleggolovkovuss** [comment]
Same for me, none of those 5 that you mentioned, only these 2

**Comment by DiAifU** [comment]
Wait, you're right. I had this message two but only once and then the two I mentionned !
`CleanupDefunctSiloEntries operation is not supported by the current implementation of IMembershipTable. Disabling the timer now.`

**Comment by ReubenBond** [comment]
Just to be clear: this was fixed in 3.1.4 by #6415, so if anyone still sees it afterwards, then please open a new issue, referencing this one and we'll take a look :)
